File: gradle/packages/goodbye-world/src/main/kotlin/com/prog/goodbye/world/GoodbyeWorld.kt
package com.prog.goodbye.world

/**
 * # `printGoodbye`
 *
 * This function prints a goodbye message to the console. If the user provides a
 * name as an argument, the function will print a goodbye message with the
 * provided name. If no argument is provided, the function will print a generic
 * goodbye message.
 *
 * __Arguments:__
 *
 * * `args: Array<String>` - An array of strings that may contain the user's name.
 *
 * __Return:__
 *
 * * `String` - A goodbye message.
 *
 * __Example:__
 *
 * ```kotlin
 * import com.prog.hello.world.printGoodbye
 * println(printGoodbye(arrayOf()))
 * println(printGoodbye(arrayOf("John", "Doe")))
 * ```
 *
 * __Output:__
 *
 * ```kotlin
 * "Goodbye World from Gradle !"
 * "Goodbye John Doe from Gradle !"
 * ```
 */
fun printGoodbye(args: Array<String>): String {
    return if (args.isNotEmpty()) {
        val name = args.drop(0).joinToString(" ")
        "Goodbye $name from Gradle !"
    } else {
        "Goodbye World from Gradle !"
    }
}

/**
 * # `printBanner`
 *
 * This function prints a banner to the console.
 *
 * __Return:__
 *
 * * String` - A banner string.
 *
 * __Example:__
 *
 * ```kotlin
 * import com.prog.hello.world.printBanner
 * println(printBanner())
 * ```
 */
fun printBanner(): String {
    return """
        |
        |     ▄████ ██▀███  ▄▄▄     ▓█████▄ ██▓   ▓█████
        |    ██▒ ▀█▓██ ▒ ██▒████▄   ▒██▀ ██▓██▒   ▓█   ▀
        |   ▒██░▄▄▄▓██ ░▄█ ▒██  ▀█▄ ░██   █▒██░   ▒███
        |   ░▓█  ██▒██▀▀█▄ ░██▄▄▄▄██░▓█▄   ▒██░   ▒▓█  ▄
        |   ░▒▓███▀░██▓ ▒██▒▓█   ▓██░▒████▓░██████░▒████▒
        |    ░▒   ▒░ ▒▓ ░▒▓░▒▒   ▓▒█░▒▒▓  ▒░ ▒░▓  ░░ ▒░ ░
        |     ░   ░  ░▒ ░ ▒░ ▒   ▒▒ ░░ ▒  ▒░ ░ ▒  ░░ ░  ░
        |   ░ ░   ░  ░░   ░  ░   ▒   ░ ░  ░  ░ ░     ░
        |         ░   ░          ░  ░  ░       ░  ░  ░  ░
        |                            ░
        |
        """.trimMargin()
}

File: csharp/src/Goodbye.World/GoodbyeWorld.cs
namespace Prog.Goodbye.World;

public class GoodbyeWorld
{
    /// <summary> <h1><c>PrintGoodbye</c></h1>
    /// <br/><br/>This method prints the goodbye message.<br/><br/>
    ///
    /// <b>Arguments:</b>
    ///
    /// <list type="bullet">
    /// <item>
    /// <c>args: string[]</c> - The command line arguments passed to the
    /// program.
    /// </item>
    /// </list>
    ///
    /// <b>Return:</b>
    ///
    /// <list type="bullet">
    /// <item>
    /// <c>string</c> - The goodbye message.
    /// </item>
    /// </list>
    ///
    /// <b>Example:</b>
    ///
    /// <code>
    /// using Prog.Goodbye.World;
    /// Console.WriteLine(GoodbyeWorld.PrintGoodbye([]));
    /// Console.WriteLine(GoodbyeWorld.PrintGoodbye(["John", "Doe"]));</code>
    ///
    /// <b>Output:</b>
    ///
    /// <code>
    /// "Goodbye World from C# !"
    /// "Goodbye John Doe from C# !"</code>
    /// </summary>
    public static string PrintGoodbye(string[] args)
    {
        if (args.Length > 0)
        {
            string name = string.Join(" ", args);
            return $"Goodbye {name} from C# !";
        }
        else
        {
            return "Goodbye World from C# !";
        }
    }

    /// <summary> <h1><c>PrintBanner</c></h1>
    /// <br/><br/>This method prints the goodbye banner.<br/><br/>
    ///
    /// <b>Return:</b>
    ///
    /// <list type="bullet">
    /// <item>
    /// <c>string</c> - The goodbye banner.
    /// </item>
    /// </list>
    /// </summary>
    public static string PrintBanner()
    {
        return @"
        ███▄    █▓████▄▄▄█████▓
        ██ ▀█   █▓█   ▓  ██▒ ▓▒
       ▓██  ▀█ ██▒███ ▒ ▓██░ ▒░
       ▓██▒  ▐▌██▒▓█  ░ ▓██▓ ░
   ██▓ ▒██░   ▓██░▒████▒▒██▒ ░
   ▒▓▒ ░ ▒░   ▒ ▒░░ ▒░ ░▒ ░░
   ░▒  ░ ░░   ░ ▒░░ ░  ░  ░
   ░      ░   ░ ░   ░   ░
    ░           ░   ░  ░
    ░
".Replace("\r\n", "\n");
    }
}
